lib/lan_connection.js

```diff
@@ -130,21 +130,29 @@ Scanner.prototype.start = function() {
       try {
         // Start discovering Tessels
         self.browser.discover();
-      } catch (error) {
-        self.emit('error', error);
+      } catch (err) {
+        process.nextTick(function() {
+          self.emit('error', err);
+        });
       }
     });
 
     // When the browser finds a new device
     self.browser.on('update', function(data) {
-      // Check if it's a Tessel
-      if (_.findIndex(self.discovered, data) > -1) {
-        return;
+      try {
+        // Check if it's a Tessel
+        if (_.findIndex(self.discovered, data) > -1) {
+          return;
+        }
+
+        var connection = new LANConnection(data);
+
+        self.emit('connection', connection);
+      } catch (err) {
+        process.nextTick(function() {
+          self.emit('error', err);
+        });
```

Comment:
Neither call to `self.emit('error'...)` should be wrapped in a nextTick, because they already occur within `setImmediate`, so any calling code has already had a chance to create an `error` event handler in the last event turn.